source/event_stream.c

```diff
@@ -749,111 +750,283 @@ struct aws_event_stream_header_value_pair aws_event_stream_create_int32_header(
     return header;
 }
 
-int aws_event_stream_add_byte_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
+int aws_event_stream_add_bool_header_by_cursor(
```

Comment:
utterly trivial: I'm betting we'd get better diffs, where it's clear the old function is calling into the new, and the new function takes over for the old, if the code was laid out like this:

- old-function, at same place in file, calls into new-cursor-function
- followed immediately by new-cursor-function, mostly using the body from the old-function
i.e.:

```
int aws_event_stream_add_byte_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
    struct aws_byte_cursor name_cursor = {
        .ptr = (uint8_t *)name,
        .len = (size_t)name_len,
    };

    return aws_event_stream_add_byte_header_by_cursor(headers, name_cursor, value);
}

int aws_event_stream_add_bool_header_by_cursor(
    struct aws_array_list *headers,
    struct aws_byte_cursor name,
    bool value) {

    AWS_FATAL_PRECONDITION(headers);
    ...mostly the same body of the old function...
```

Reply:
Not sure it is any better, but went ahead and did so